agent/eventhandler/handler.go

```diff
@@ -30,27 +30,20 @@ func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, save
 	statesaver = saver
 
 	for {
-		taskEvents, containerEvents := taskEngine.TaskEvents()
+		stateChangeEvents := taskEngine.StateChangeEvents()
 
-		for taskEvents != nil && containerEvents != nil {
+		for stateChangeEvents != nil {
 			select {
-			case event, open := <-containerEvents:
-				if !open {
-					containerEvents = nil
-					log.Error("Container events closed")
+			case event, ok := <-stateChangeEvents:
+				if !ok {
```

Comment:
so from the [go spec](https://golang.org/ref/spec#Receive_operator) the boolean result from the channel receive operator can be "false if it is a zero value generated because the channel is closed and empty."